### PR TITLE
feat: create POST/ login Api and its tests

### LINKED ISSUE #120: [backend] Create Login API

### PR DESCRIPTION
**Pull request by mohammedagha27**
close #120

### CODE REVIEW

File: server/src/helpers/jwt.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import { Payload } from '../interfaces';
+
+dotenv.config();
```
Comment (LinaYahya) [inline server/src/helpers/jwt.ts:5]:
```suggestion
```
No need to import it every where as import it at the app level will expose it where we need

File: server/src/controllers/authentication.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { User } from '../models';
+import { loginValidation } from '../validation';
+import { generateToken } from '../helpers';
+
+const loginMiddleware = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+  await loginValidation.validate({ email, password });
+
+  const user = await User.findOne({ where: { email } });
+  if (!user) return { status: 400, msg: 'User not found' };
```
Comment (LinaYahya) [inline server/src/controllers/authentication.ts:12]:
```suggestion
  if (!user) throw serverErrs.BadReq
```
Also, I'd prefer to send hazy message, that email or password incorrect
Read more on [Authentication and error messages](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#authentication-and-error-messages)

File: server/src/controllers/authentication.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Request, Response } from 'express';
+import { compare } from 'bcrypt';
+import { User } from '../models';
+import { loginValidation } from '../validation';
+import { generateToken } from '../helpers';
+
+const loginMiddleware = async (req: Request, res: Response) => {
```
Comment (LinaYahya) [inline server/src/controllers/authentication.ts:7]:
```suggestion
const login = async (req: Request, res: Response) => {
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by naghamabuwarda — APPROVED** [review]

**Review by shathakh — APPROVED** [review]

File: server/src/controllers/authentication.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const user = await User.findOne({ where: { email } });
+  if (!user) throw serverErrs.BAD_REQUEST('Wrong Email Or Password');
+  const result = await compare(password, user.password);
+
+  if (!result) throw serverErrs.BAD_REQUEST('Wrong Email Or Password');
+
+  const { name, role, id } = user;
+  const token = await generateToken({ userID: id, role, name });
+  res.cookie('token', token);
+  return { status: 200, msg: 'logged in successfully' };
```
Comment (LinaYahya) [inline server/src/controllers/authentication.ts:20]: I think we will need to return values needed to be set at `UserContext`, For now we will leave it and we can fix when we send request at login page

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**